pkg/autoupdate/tuf/autoupdate.go

```diff
@@ -164,73 +163,90 @@ func (ta *TufAutoupdater) checkForUpdate() error {
 		return fmt.Errorf("could not get complete list of targets: %w", err)
 	}
 
-	targetReleaseFile := fmt.Sprintf("%s/%s/%s/release.json", ta.binary, ta.operatingSystem, ta.channel)
+	for _, binary := range []string{"launcher", "osqueryd"} {
+		if err := ta.findRelease(binary, targets); err != nil {
+			return fmt.Errorf("could not find release: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (ta *TufAutoupdater) findRelease(binary string, targets data.TargetFiles) error {
+	targetReleaseFile := fmt.Sprintf("%s/%s/%s/release.json", binary, ta.operatingSystem, ta.channel)
 	for targetName, target := range targets {
 		if targetName != targetReleaseFile {
 			continue
 		}
 
 		// We found the release file that matches our OS and binary. Evaluate it
 		// to see if we're on this latest version.
-		type releaseFileCustomMetadata struct {
-			Target string `json:"target"`
-		}
-
-		var custom releaseFileCustomMetadata
+		var custom ReleaseFileCustomMetadata
 		if err := json.Unmarshal(*target.Custom, &custom); err != nil {
 			return fmt.Errorf("could not unmarshal release file custom metadata: %w", err)
 		}
 
 		level.Debug(ta.logger).Log(
 			"msg", "checked most up-to-date release from TUF",
 			"launcher_version", version.Version().Version,
-			"release_version", ta.versionFromTarget(custom.Target),
-			"binary", ta.binary,
+			"release_version", ta.versionFromTarget(custom.Target, binary),
+			"binary", binary,
 			"channel", ta.channel,
 		)
 
-		break
+		return nil
 	}
 
-	return nil
+	return fmt.Errorf("expected release file %s for binary %s to be in targets but it was not", targetReleaseFile, binary)
 }
 
-func (ta *TufAutoupdater) versionFromTarget(target string) string {
+func (ta *TufAutoupdater) versionFromTarget(target string, binary string) string {
 	// The target is in the form `launcher/linux/launcher-0.13.6.tar.gz` -- trim the prefix and the file extension to return the version
-	prefixToTrim := fmt.Sprintf("%s/%s/%s-", ta.binary, ta.operatingSystem, ta.binary)
+	prefixToTrim := fmt.Sprintf("%s/%s/%s-", binary, ta.operatingSystem, binary)
 
 	return strings.TrimSuffix(strings.TrimPrefix(target, prefixToTrim), ".tar.gz")
 }
 
-func (ta *TufAutoupdater) rollingErrorCount() int {
-	ta.lock.RLock()
-	defer ta.lock.RUnlock()
+func (ta *TufAutoupdater) storeError(autoupdateErr error) {
+	ta.lock.Lock()
+	defer ta.lock.Unlock()
```

Comment:
do we still need a lock here at all? I think we can rely on the underlying storage to be thread safe.

Reply:
Ah yup you're right! I will triple-check and then remove 👍